### ISSUE TITLE
Lndbtc/lndltc don't connect to cloud instances on boot

### ISSUE THREAD

**Issue by kilrau** [ExchangeUnion/xud-docker#91]
Sometimes `lndltc` doesn't connect to cloud instances immediately after being up. Output below shows that `lndltc` did NOT connect to our cloud instance, hence doesn't get a channel.

```
simnet > getinfo
D0722 12:10:59.553098618     627 env_linux.cc:71]            Warning: insecure environment read function 'getenv' used
{
  "version": "1.0.0-testnet.1",
  "nodePubKey": "0245c9daa8a38849073cdcf8fcb7b58e979443535d5902aa1fac0e5cb5c6b79c47",
  "urisList": [
    "0245c9daa8a38849073cdcf8fcb7b58e979443535d5902aa1fac0e5cb5c6b79c47@0.0.0.0:8885"
  ],
  "numPeers": 3,
  "numPairs": 4,
  "orders": {
    "peer": 10,
    "own": 0
  },
  "lndMap": [
    [
      "BTC",
      {
        "error": "",
        "channels": {
          "active": 0,
          "inactive": 0,
          "pending": 1
        },
        "chainsList": [
          {
            "chain": "bitcoin",
            "network": "simnet"
          }
        ],
        "blockheight": 30700,
        "urisList": [],
        "version": "0.6.1-beta commit=v0.6.1-beta-dirty",
        "alias": "lndbtc"
      }
    ],
    [
      "LTC",
      {
        "error": "",
        "channels": {
          "active": 0,
          "inactive": 0,
          "pending": 0
        },
        "chainsList": [
          {
            "chain": "litecoin",
            "network": "simnet"
          }
        ],
        "blockheight": 15042,
        "urisList": [],
        "version": "0.6.1-beta commit=v0.6.1-beta-dirty",
        "alias": "lndltc"
      }
    ]
  ],
  "raiden": {
    "error": "",
    "address": "0x58328199729fe36ea1a8dE812983c069F9EbC64C",
    "channels": 2,
    "version": ""
  }
}
simnet > lndltc-lncli getinfo
{
	"version": "0.6.1-beta commit=v0.6.1-beta-dirty",
	"identity_pubkey": "0267c5764395950c6ff805e52c672802f50c3cf0570086b0da4baa1ee837c6a8e2",
	"alias": "lndltc",
	"num_pending_channels": 0,
	"num_active_channels": 0,
	"num_inactive_channels": 0,
	"num_peers": 0,
	"block_height": 15042,
	"block_hash": "ea2dba6dbbe61775d2a95e33f6b351a0146ed62d8944fd35e79aa6f91418d60d",
	"best_header_timestamp": 1563797416,
	"synced_to_chain": true,
	"testnet": false,
	"chains": [
		{
			"chain": "litecoin",
			"network": "simnet"
		}
	],
	"uris": null
}
simnet > lndltc-lncli listpeers
{
    "peers": [
    ]
}
simnet > lndbtc-lncli listpeers
{
    "peers": [
        {
            "pub_key": "03a0bdf210dcecebb2d6f092b47fb7753af8114261888a365dfcb0d4548bbbdd2a",
            "address": "35.196.118.79:10012",
            "bytes_sent": "686",
            "bytes_recv": "14937",
            "sat_sent": "0",
            "sat_recv": "0",
            "inbound": false,
            "ping_time": "107880",
            "sync_type": "ACTIVE_SYNC"
        }
    ]
}
```

**Comment by kilrau** [comment]
I found the reason:

When trying to connect manually to the cloud, `lndltc` throws the error 
```
simnet > lndltc-lncli connect 0270e8254e07649cdde230c9e09de6ff63c28a6d275c30b0a6863028d9db0e7c6f@xud1.test.exchangeunion.com:9735
[lncli] rpc error: code = Unknown desc = chain backend is still syncing, server not active yet
```

which is not true. LTCD is caught up with the chain and `lndltc` shows `"synced_to_chain": true`:

```
simnet > lndltc-lncli getinfo
{
	"version": "0.6.1-beta commit=v0.6.1-beta-dirty",
	"identity_pubkey": "0267c5764395950c6ff805e52c672802f50c3cf0570086b0da4baa1ee837c6a8e2",
	"alias": "lndltc",
	"num_pending_channels": 0,
	"num_active_channels": 0,
	"num_inactive_channels": 0,
	"num_peers": 0,
	"block_height": 15049,
	"block_hash": "ba41bea18175ce0485fc02c81b2ab7d80bc53a59fa7a17ba7d9114f8ceecc09e",
	"best_header_timestamp": 1563797863,
	"synced_to_chain": true,
	"testnet": false,
	"chains": [
		{
			"chain": "litecoin",
			"network": "simnet"
		}
	],
	"uris": null
}
```

Waiting for 10+ minutes resolves this, all commands work and `lndltc` finally connect to our cloud instance:
```
simnet > lndltc-lncli listpeers
{
    "peers": [
        {
            "pub_key": "03e26345aa5d7024668e1f16adc2ecd85907a930820ad5a32736946cb824067eeb",
            "address": "35.229.81.83:10011",
            "bytes_sent": "572",
            "bytes_recv": "5757",
            "sat_sent": "0",
            "sat_recv": "0",
            "inbound": false,
            "ping_time": "108296",
            "sync_type": "ACTIVE_SYNC"
        }
    ]
}
```

**Comment by kilrau** [comment]
Proposal what to do: move to litecoin core.

**Comment by kilrau** [comment]
I tested a series of fresh simnet setups today. Native, as well as docker. On both I face the issue of lnd's not connecting to our cloud lnd's (lncli listpeers empty) and hence not getting a channel. Manual connect also doesn't work in these cases:
```simnet > lndltc-lncli connect 0270e8254e07649cdde230c9e09de6ff63c28a6d275c30b0a6863028d9db0e7c6f@xud1.simnet.exchangeunion.com:10735
[lncli] rpc error: code = Unknown desc = dial tcp 35.196.118.79:10735: connect: connection refused
simnet > lndbtc-lncli connect 03a0bdf210dcecebb2d6f092b47fb7753af8114261888a365dfcb0d4548bbbdd2a@xud1.simnet.exchangeunion.com:9735
[lncli] rpc error: code = Unknown desc = dial tcp 35.196.118.79:9735: connect: connection refused
simnet > lndltc-lncli connect 0270e8254e07649cdde230c9e09de6ff63c28a6d275c30b0a6863028d9db0e7c6f@xud1.simnet.exchangeunion.com:9735
[lncli] rpc error: code = Unknown desc = dial tcp 35.196.118.79:9735: connect: connection refused```

**Comment by michael1011** [comment]
@kilrau was this fixed for you by the upgrade to LND 0.7.1?

**Comment by kilrau** [comment]
Yes, don't see this anymore. Closing for now. If it pops up again, I'll reopen.
